addons/partner_autocomplete/models/res_partner.py

```diff
@@ -181,3 +181,21 @@ def write(self, values):
             self._update_autocomplete_data(values.get('vat', False))
 
         return res
+
+    @api.model
+    def _get_view(self, view_id=None, view_type='form', **options):
+        arch, view = super()._get_view(view_id, view_type, **options)
+        nodes_to_edit = []
+
+        if view in (self.env.ref('base.view_partner_form'), self.env.ref('base.view_partner_simple_form')):
+            nodes_to_edit.extend(arch.xpath("//div[hasclass('oe_title')]/h1/field[@id='company']"))
+            nodes_to_edit.extend(arch.xpath("//div[hasclass('oe_title')]/h1/field[@id='individual']"))
+            nodes_to_edit.extend(arch.xpath("//field[@name='vat']"))
+        elif view == self.env.ref('base.view_company_form'):
+            nodes_to_edit.extend(arch.xpath("//div[hasclass('oe_title')]/h1/field[@name='name']"))
+            nodes_to_edit.extend(arch.xpath("//field[@name='vat']"))
+
+        for node in nodes_to_edit:
+            node.attrib['widget'] = 'field_partner_autocomplete'
```

Comment:
What about something like this?
```suggestion
    if view_type == 'form':
        for node in arch.xpath(
            "//field[@name='vat']"
            "|//field[@name='company']"
            "|//field[@name='individual']"
            "|//field[@name='name']"
        ):
            if not node.attrib['widget']:
                node.attrib['widget'] = 'field_partner_autocomplete'
```

Or is it too aggressive?

Reply:
I went for a literal "translation" of the views in my commit, but we could indeed simplify the XPath expressions, it should make them more robust to future changes and view inheritance.
I'll adapt the commit with this version.

The widget has too be forced though, there is already a `text` widget in [some places](https://github.com/odoo/odoo/blob/saas-16.3/odoo/addons/base/views/res_partner_views.xml#L188) that would prevent adding the `field_partner_autocomplete` widget.